logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -229,9 +235,28 @@ public void stop() {
 
     @Override
     public void addAppender(Appender<ILoggingEvent> newAppender) {
+        // When SocketAppender is used and event object contains classes
+        // that are not on whitelist, HardenedObjectInputStream raises
+        // InvalidClassException: Unauthorized deserialization attempt.
+        needsHashMap = isSocketAppender(newAppender);
         aai.addAppender(newAppender);
     }
 
+    @SuppressWarnings("unchecked")
+    private boolean isSocketAppender(Appender<ILoggingEvent> appender) {
+        if (appender instanceof AppenderAttachable) {
+            boolean result = false;
+            for (Iterator<Appender<ILoggingEvent>> i = ((AppenderAttachable<ILoggingEvent>) appender)
+                    .iteratorForAppenders(); i.hasNext();) {
+                final Appender<ILoggingEvent> nested = i.next();
+                result = result || isSocketAppender(nested);
+            }
+            return result;
+        }
+
+        return appender instanceof AbstractSocketAppender;
```

Comment:
Let's do the cheap operation first:
```suggestion
        if (appender instanceof AbstractSocketAppender) {
            return true;
        }
        if (appender instanceof AppenderAttachable) {
            for (final Iterator<Appender<ILoggingEvent>> i = ((AppenderAttachable<ILoggingEvent>) appender)
                    .iteratorForAppenders(); i.hasNext();) {
                if (isSocketAppender(i.next())) {
                    return true;
                }
            }
        }

        return false;
```

Reply:
you an expert in early returns :) fixed, thanks.

Reply:
🤣